### PR TITLE
change: don't treat route segment with ':' as parameter by default

### LINKED ISSUE #3134: I have some trouble on route match
### Issue description
I set two routes like this:
##route1
{"uris":["/test/v1/files/*"],"name":"get_file1","desc":"get_file1","methods":["GET"],"vars":[],"plugins":{"proxy-rewrite":{"regex_uri":["^/test/v1/files/(.+)","/v1/files/$1"]}},"upstream_id":"333773215308464216"}
##route2
{"uris":["/test/v1/files:listReputationHistories"],"name":"list_reputation_history1","methods":["POST"],"vars":[],"plugins":{"proxy-rewrite":{"uri":"/v1/files:listReputationHistories"}},"upstream_id":"333773215308464216"}

But run command: `curl "http://127.0.0.1:9080/test/v1/files:xxxdsadsadxxx" -XPOST -d {"uri":"xxx"}` , It also can return results.
APISIX sames to match '/test/v1/files:xxxdsadsadxxx' to '"/test/v1/files:listReputationHistories'. I also try other cases: `curl "http://127.0.0.1:9080/test/v1/files: {any string}" -XPOST -d {"uri":"xxx"}` , APISIX sames ignore string behind ':'.

### Environment

* apisix version (cmd: `2.1`):

### PR DESCRIPTION
**Pull request by spacewander**
Fix #3134.

Signed-off-by: spacewander <spacewanderlzx@gmail.com>

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

**Comment by spacewander** [comment]
Another related issue: https://github.com/apache/apisix/issues/2008

**Review by membphis — COMMENTED** [review]
In this case, how about creating a new router?

eg: `radixtree_uri`, `radixtree_uri_no_parameter`

This configuration is not easy to understand and find. 

https://github.com/apache/apisix/pull/3154/files#diff-8d872babc717e9d733641b56bfc530ef98751fbe4e68f08d79b2b83109c22fffR122-R123

**Review by tokers — APPROVED** [review]

File: conf/config-default.yaml [hunk 1]
```diff
@@ -119,6 +119,8 @@ apisix:
   # control:
   #   ip: "127.0.0.1"
   #   port: 9090
+  feature:
+    enable_parameter_in_route: false        # allow using ':param' in the router. See https://github.com/api7/lua-resty-radixtree/#parameters-in-path
```
Comment (membphis) [inline conf/config-default.yaml:123]: We need to find a better way

**Review by membphis — CHANGES_REQUESTED** [review]

**Comment by spacewander** [comment]
We decide to use `radixtree_uri_with_parameter` instead after discussion.

File: conf/config-default.yaml (hunk 1 above)
Comment (spacewander) [reply conf/config-default.yaml]: @membphis 
rewritten.

**Comment by membphis** [comment]
need to add some comment here:

https://github.com/apache/apisix/blob/master/conf/config-default.yaml#L89-L91

**Comment by spacewander** [comment]
@membphis 
Added.

**Review by membphis — APPROVED** [review]
LGTM

**Comment by membphis** [comment]
many thx, merged

**Pull request merged**